react/src/components/ServiceLauncherModal.tsx

```diff
@@ -151,27 +237,118 @@ const ServiceLauncherModal: React.FC<ServiceLauncherProps> = ({
+      $endpoint_id: UUID!
+      $props: ModifyEndpointInput!
+    ) {
+      modify_endpoint(endpoint_id: $endpoint_id, props: $props) {
+        ok
+        msg
+        endpoint {
+          endpoint_id
+          desired_session_count
+          resource_group
+          resource_slots
+          resource_opts
+          cluster_mode
+          cluster_size
+          open_to_public
+          image
+          architecture
+          name
+        }
+      }
+    }
+  `);
+
   // Apply any operation after clicking OK button
   const handleOk = () => {
     form
       .validateFields()
       .then((values) => {
-        mutationToCreateService.mutate(values, {
-          onSuccess: () => {
-            onRequestClose(true);
-          },
-          onError: (error) => {
-            if (error?.message) {
-              message.error(
-                _.truncate(error?.message, {
-                  length: 200,
+        if (endpoint) {
+          const mutationVariables = {
+            endpoint_id: endpoint?.endpoint_id || '',
+            props: {
+              resource_slots: JSON.stringify({
+                cpu: values.resource.cpu.toString(),
+                mem: values.resource.mem,
+                ...(values.resource.accelerator > 0
+                  ? {
+                      [values.resource.acceleratorType]:
+                        values.resource.accelerator,
+                    }
+                  : undefined),
+              }),
+              resource_opts: JSON.stringify({ shmem: values.resource.shmem }),
+              // FIXME: temporally convert cluster mode string according to server-side type
+              cluster_mode:
+                'single-node' === values.cluster_mode
+                  ? 'SINGLE_NODE'
+                  : 'MULTI_NODE',
+              cluster_size: values.cluster_size,
+              desired_session_count: values.desiredRoutingCount,
+              ...getImageInfoFromInputInEditing(
+                checkManualImageAllowed(
+                  baiClient._config.allow_manual_image_name_for_session,
+                  values.environments?.manual,
+                ),
+                values,
+              ),
+              name: values.serviceName,
+              resource_group: values.resourceGroup,
+            },
+          };
+          commitModifyEndpoint({
+            variables: mutationVariables,
+            onCompleted: (res, error) => {
+              const updatedEndpoint = res.modify_endpoint?.endpoint;
+              message.success(
+                t('modelService.ServiceUpdated', {
+                  name: updatedEndpoint?.name,
                 }),
               );
-            } else {
-              message.error(t('modelService.FailedToStartService'));
-            }
-          },
-        });
+              if (error) {
+                message.error(t('dialog.ErrorOccurred'));
+              } else {
+                onRequestClose(true);
+              }
```

Comment:
When an error exists, both message.success and message.error are executed.
```suggestion
              if (error) {
                message.error(t('dialog.ErrorOccurred'));
              } else {
                message.success(
                  t('modelService.ServiceUpdated', {
                    name: updatedEndpoint?.name,
                  }),
                );
                onRequestClose(true);
              }
```

Reply:
<img width="547" alt="image" src="https://github.com/lablup/backend.ai-webui/assets/621215/82bca2eb-3e08-4037-9c4b-4ab0aa6d7bbb">